test_get_nam.sh

```diff
@@ -77,29 +98,140 @@ sed -e "s/%NULLNAMWINPREDATAPATH%/$escSCENARIODIR/" \
     -e "s/\"%NULLAPPLYRAMP%\"/$boolApplyRamp/" \
     -e "s/\"%NULLRAMPDIST%\"/$SPATIALEXTRAPOLATIONRAMPDISTANCE/" \
      < get_nam_data.pl.json \
-     | $SCRIPTDIR/NAMtoOWIRamp.pl > /dev/null
-preFile=$($SCRIPTDIR/metadata.pl --key winPrePressureFile < NAMtoOWIRamp.pl.json)
-winFile=$($SCRIPTDIR/metadata.pl --key winPreVelocityFile < NAMtoOWIRamp.pl.json)
-mv $winFile $preFile fort.22 $SCENARIODIR
-mv get_nam_data.pl.* NAMtoOWIRamp.pl.* $SCENARIODIR
+     | $SCRIPTDIR/NAMtoOWIRamp.pl > NAMtoOWIRamp.pl.json
+preFile=$($SCRIPTDIR/bin/bashJSON.pl --key winPrePressureFile < NAMtoOWIRamp.pl.json)
+winFile=$($SCRIPTDIR/bin/bashJSON.pl --key winPreVelocityFile < NAMtoOWIRamp.pl.json)
+mv fort.22 $SCENARIODIR
+mv NAMtoOWIRamp.pl.* get_nam_status.pl.* asgs_main.sh_get_nam_status.json rotatedNAM.txt select_nam_nowcast.pl.json reproject.log lambert_diag.out $SCENARIODIR
+cp get_nam_data.pl.* $SCENARIODIR  # needed for forecast
 #  f o r e c a s t
 stage=FORECAST
 SCENARIO=namforecast
 SCENARIODIR=$RUNDIR/namforecast
 if [[ ! -d $SCENARIODIR ]]; then mkdir $SCENARIODIR ; fi
 escSCENARIODIR=${SCENARIODIR////'\/'}
+sed "s/NOWCAST/FORECAST/" < get_nam_data.pl.json > get_nam_forecast.json
+if [[ $? != 0 ]]; then
+    echo "ERROR: Failed to replace NOWCAST with FORECAST in select_nam_nowcast.pl.json with sed."
+    exit
+fi
+# download the forecast data
+get_nam_data.pl < get_nam_forecast.json > get_nam_data.pl.json
+if [[ $? != 0 ]]; then
+    echo "ERROR: Failed to download NAM forecast data with get_nam_data.pl."
+    exit
+fi
+# NAMtoOWIRamp.pl writes the ascii OWI WIN/PRE files to
+# the scenario directories
+sed -e "s/%NULLNAMWINPREDATAPATH%/$escSCENARIODIR/" \
+    -e "s/%NULLNAMWINPREGRID%/$escPtFilePath/" \
+    -e "s/\"%NULLNAMAWIPGRID%\"/218/" \
+    -e "s/%NULLNAMRAWFORMAT%/grib2/" \
+    -e "s/\"%NULLVELMULT%\"/$VELOCITYMULTIPLIER/" \
+    -e "s/\"%NULLPRESSMULT%\"/0.01/" \
+    -e "s/\"%NULLAPPLYRAMP%\"/$boolApplyRamp/" \
+    -e "s/\"%NULLRAMPDIST%\"/$SPATIALEXTRAPOLATIONRAMPDISTANCE/" \
+     < get_nam_data.pl.json \
+     | $SCRIPTDIR/NAMtoOWIRamp.pl > NAMtoOWIRamp.pl.json
+preFile=$($SCRIPTDIR/bin/bashJSON.pl --key winPrePressureFile < NAMtoOWIRamp.pl.json)
+winFile=$($SCRIPTDIR/bin/bashJSON.pl --key winPreVelocityFile < NAMtoOWIRamp.pl.json)
+mv fort.22 $SCENARIODIR
+mv get_nam_forecast.json get_nam_data.pl.* NAMtoOWIRamp.pl.* rotatedNAM.txt reproject.log lambert_diag.out forecast.properties $SCENARIODIR
+#
+#  N O W C A S T   F R O M   F I L E S Y S T E M
+#
+# test the construction of a nowcast from grib2 files already
+# on the filesystem
+# set variables to be used in json template
+RUNDIR=$PWD
+namArchiveDir=$RUNDIR/nam-filesystem
+if [[ -d $namArchiveDir ]]; then
+    rm -rf $namArchiveDir
+fi
+# use the files already downloaded to construct a nowcast
+# from the filesystem
+mv $instanceNamDir $namArchiveDir
+BACKSITE=filesystem          # <--
```

Comment:
What do `# <--` indicate?

Reply:
Those arrows are a note-to-self to indicate the differences in the input parameters for the test for nowcast constructed from files on the filesystem vs the test for input parameters for a nowcast constructed by downloading files from NCEP. I wanted to make sure I minimized and rationalized the number of parameters that would need to change.

Reply:
thank you!